squid-forward-proxy-operator/src/charm.py

```diff
@@ -89,35 +117,77 @@ def _reconcile(self, _: ops.EventBase) -> None:
             status = ops.BlockedStatus
             status_message += f", integration errors: {integration_errors}"
         self.unit.status = status(status_message)
+        self._set_peer_data(
+            proxy_requests=proxy_requests,
+            proxy_users=proxy_users,
+            status=status.name,
+            status_message=status_message,
+        )
 
-    def _update_config_passwd(
-        self, proxy_requests: list[http_proxy.HttpProxyRequest], proxy_users: dict[str, str]
+    def _set_peer_data(
+        self,
+        proxy_requests: list[http_proxy.HttpProxyRequest],
+        proxy_users: dict[str, str],
+        status: str,
+        status_message: str,
     ) -> None:
-        """Update Squid configuration and passwd file.
+        """Set replica consensus data in the peer integrations.
 
         Args:
-            proxy_requests: list of http proxy requests
-            proxy_users: required http proxy users
+            proxy_requests: http proxy requests.
+            proxy_users: http proxy users
+            status: unit status
+            status_message: unit status message
         """
-        old_config = self._squid.read_config()
-        old_passwd = self._squid.read_passwd()
-        new_config = self._squid.generate_config(
-            specs=proxy_requests,
-            http_port=self.config["http-port"],
+        integration = typing.cast(
+            ops.Relation, self.model.get_relation(relation_name=PEER_INTEGRATION_NAME)
+        )
+        integration_data = integration.data[self.app]
+        integration_data["proxy-requests"] = json.dumps(
+            [r.model_dump(mode="json") for r in proxy_requests]
+        )
+        secret_id = integration_data.get("proxy-users")
+        if secret_id:
+            secret = self.model.get_secret(id=secret_id)
+            content = secret.get_content(refresh=True)
+            if json.loads(content["data"]) != proxy_users:
+                secret.set_content({"data": json.dumps(proxy_users)})
+        else:
+            secret = self.app.add_secret(content={"data": json.dumps(proxy_users)})
+            integration_data["proxy-users"] = typing.cast(str, secret.id)
```

Comment:
Maybe add a method to expose the secret ID from the library and just pass it directly in the peer relation

Reply:
Changed to store the list of juju secret IDs in the peer integration.